src/Connection.ts

```diff
@@ -275,22 +285,36 @@ const DEFAULT_MAX_RETRIES = 10
  */
 
 export default class Connection extends EventEmitter {
+
+    _debug: Todo
+    options: Todo
+    retryCount: Todo
+    wantsState: Todo
+    connectionHandles: Todo
+    step: Todo
+    socket?: Todo
+    didDisableAutoConnect?: Todo
+    isWaiting?: Todo
+    _isReconnecting: Todo
+    _backoffTimeout: Todo
+    sendID: Todo
+
     static getOpen() {
         return openSockets.size
     }
 
     static async closeOpen() {
-        return Promise.all([...openSockets].map(async (socket) => {
+        return Promise.all([...openSockets].map(async (socket: Todo) => {
             socket[FORCE_CLOSED] = true // eslint-disable-line no-param-reassign
             return CloseWebSocket(socket).catch((err) => {
                 socket.debug(err) // ignore error
             })
         }))
     }
 
-    constructor(options = {}, client) {
+    constructor(options = {}, debug: Debug.Debugger) {
```

Comment:
Note original code worked like this:
https://github.com/streamr-dev/streamr-client-javascript/pull/206/commits/f3bae1382eb6d651d705195168ce8d71008212eb#diff-0e149db9f5c84dd5e3126b72e3cdcf3c297e6f2836d6a1b6bf0dea41cf05760fL66-L71

```js
        if (options.debug) {	
            this._debug = options.debug.extend(id)	
        } else {	
            this._debug = Debug(`StreamrClient::${id}`)	
        }
```

`debug` should be optional.

Reply:
Changed that to an optional parameter. 

In the previous PR I marked `debug` as a required parameter because I looked like the `Connection` and `StreamrClient` debug instances are in practice linked together (in production code `Connection`'s debug was always derived from `StreamrClient`'s debug). But maybe optionality better as it makes `Connection` more independent in theory.